EOS/IOUtils.py

```diff
@@ -0,0 +1,69 @@
+import hashlib
+import os
+import glob
+import traceback
+import re
+import binascii
+
+BLOCKSIZE = 65536
+
+class ReadWriteOp:
+
+    def __init__(self):
+        pass
+
+    def convert_bytes(self, num):
+        """
+        this function will convert bytes to MB.... GB... etc
+        """
+        for x in ['bytes', 'KB', 'MB', 'GB', 'TB']:
+            if num < 1024.0:
+                return "%3.1f %s" % (num, x)
+            num /= 1024.0
+
+
+    def extract_digits(self, s):
+        """
+        this function separates digits and strings
+        """
+        string = s.strip('0123456789')
+        digit = int(re.search(r'\d+', s).group(0))
+        return digit, string
+
+
+    def convert_size(self, string):
+        """
+        this function calculates total bytes
+        """
+        units = ['K', 'M', 'G']
+        requested_size, unit = self.extract_digits(string)
+        # for conversion of lowercase to uppercase
+        if (unit.islower()):
+            unit = unit.upper()
+        byte = 1
+        if unit in units:
+            for i in range(units.index(unit) + 1):
+                byte *= 1024
+        return requested_size, byte
+
+
+
+class Checksum:
+    def __init__(self):
+        self.source_checksum = None
+        self.dest_checksum = None
+
+    def calculate_source_hash(self, data):
```

Comment:
Collapse calculate_source_hash and calculate_destination_hash into a single function calculate_hash.
It gets whatever content as parameter (e.g., the payload to be written to file or the payload read-back from a file) and returns its hash